### DISCUSSION TITLE
Integrations with security tools

### DISCUSSION THREAD

**Discussion by lizrice** [aquasecurity/starboard#104] [Development]
Our initial release of Starboard includes integrations with
* [Trivy](https://github.com/aquasecurity/trivy) for vulnerability scanning
* [Polaris](https://github.com/fairwindsops/polaris) for config auditing
* [kube-bench](https://github.com/aquasecurity/kube-bench) for Kubernetes configuration checks
* [kube-hunter](https://github.com/aquasecurity/kube-hunter) for basic Kubernetes pen-testing

What other tools would you like to see integrated with Starboard?

**Comment by danielpacak** [comment]
Related to that list is the idea to integrate [Popeye](https://github.com/derailed/popeye). See #300 for more details.

**Comment by bgoareguer** [comment]
[conftest](https://github.com/open-policy-agent/conftest) could be a nice addition considering the work done in [appshield](https://github.com/aquasecurity/appshield/)

**Reply by danielpacak** [reply]
That's a great idea @bgoareguer and I'm actually working on a POC to evaluate such integration.

**Comment by ragashreeshekar** [comment]
I would like to see integrations such as dynamic scan tools such as kubei, kube-scan

**Reply by danielpacak** [reply]
👋 @ragashreeshekar Could you elaborate a bit on your proposal? IIRC kubei has it's own scan jobs scheduler (similar to Starboard) and integrates with Clair vulnerability scanner as well as Dockle to benchmark container images. I could imagine integrating Starboard with Clair and / or Dockle directly, but what do you mean by integrating Starboard with kubei?

And I have a similar question for https://github.com/octarinesec/kube-scan How would you integrate with Starboard? Is it possible to run Kube-scan as a container / Kubernetes job in a similar way to Trivy? I'm not very familiar with Kube-scan, but it looks like a self-contained product with its own dashboard and in-memory security reports storage. How would that fit into Starboard, which does workloads discovery, scan jobs scheduling, and saves reports as custom resources?

**Comment by bgoareguer** [comment]
Starboard can run the CIS benchmark for K8S thanks to kube-bench. 
It would be great if Startboard could also run the CIS benchmark for Linux (maybe using [linux-bench](https://github.com/aquasecurity/linux-bench))

Also, do you think Starboard could scan the host machines for vulnerabilities (maybe using Trivy or [Vuls](https://vuls.io/))?
